pyanaconda/bootloader.py

```diff
@@ -2303,16 +2416,27 @@ def install(self, args=None):
 
 
 # every platform that wants a bootloader needs to be in this dict
-bootloader_by_platform = {platform.X86: GRUB2,
-                          platform.EFI: EFIGRUB,
-                          platform.MacEFI: MacEFIGRUB,
-                          platform.PPC: GRUB2,
-                          platform.IPSeriesPPC: IPSeriesGRUB2,
-                          platform.NewWorldPPC: MacYaboot,
-                          platform.S390: ZIPL,
-                          platform.Aarch64EFI: Aarch64EFIGRUB,
-                          platform.ARM: EXTLINUX,
-                          platform.omapARM: EXTLINUX}
+if flags.cmdline.get("legacygrub") == "1":
+    log.info("Using legacy grub (0.9x)")
+    bootloader_by_platform = {platform.X86: GRUB,
+                              platform.EFI: EFIGRUB1,
+                              platform.MacEFI: MacEFIGRUB,
+                              platform.PPC: GRUB2,
+                              platform.IPSeriesPPC: IPSeriesGRUB2,
+                              platform.NewWorldPPC: MacYaboot,
+                              platform.S390: ZIPL,
+                              platform.Aarch64EFI: Aarch64EFIGRUB}
+else:
+    bootloader_by_platform = {platform.X86: GRUB2,
+                              platform.EFI: EFIGRUB,
+                              platform.MacEFI: MacEFIGRUB,
+                              platform.PPC: GRUB2,
+                              platform.IPSeriesPPC: IPSeriesGRUB2,
+                              platform.NewWorldPPC: MacYaboot,
+                              platform.S390: ZIPL,
+                              platform.Aarch64EFI: Aarch64EFIGRUB,
+                              platform.ARM: EXTLINUX,
+                              platform.omapARM: EXTLINUX}
 
```

Comment:
I would do this differently, setup the dict  as before and then modify the 2 entries using bootloader_by_platform.update({platform.X86: GRUB, platform.EFI: EFIGRUB1}) based on the flag.